### PR TITLE
Refactoring QuestionWidget and extenders

### PR DESCRIPTION
**Pull request by phillipm**
Classes that implement QuestionWidget weren't making use of the `mPrompt` class field, but were passing the prompt around manually. This refactor makes sure they set `mPrompt` in the constructor and just use that.

While I was doing that, I also linted all the files I touched, extracted duplicate code, and removed unused code

### CODE REVIEW

File: app/src/org/odk/collect/android/widgets/AutoCompleteWidget.java [hunk 1]
```diff
@@ -1,293 +0,0 @@
...
-/**
- * AutoCompleteWidget handles select-one fields using an autocomplete text box. The user types part
- * of the desired selection and suggestions appear in a list below. The full list of possible
- * answers is not displayed to the user. The goal is to be more compact; this question type is best
- * suited for select one questions with a large number of possible answers. If images, audio, or
- * video are specified in the select answers they are ignored.
- * 
- * @author Jeff Beorse (jeff@beorse.net)
- */
-public class AutoCompleteWidget extends QuestionWidget {
```
Comment (phillipm) [inline app/src/org/odk/collect/android/widgets/AutoCompleteWidget.java]: Unused

File: app/src/org/odk/collect/android/adapters/ImageAdapter.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+import org.commcare.android.util.MediaUtil;
+import org.commcare.android.util.StringUtils;
+import org.commcare.dalvik.R;
+import org.javarosa.core.reference.InvalidReferenceException;
+import org.javarosa.core.reference.ReferenceManager;
+
+import java.io.File;
+
+public class ImageAdapter extends BaseAdapter {
```
Comment (phillipm) [inline app/src/org/odk/collect/android/adapters/ImageAdapter.java]: This was duplicated in `GridWidget` and `GridMultiWidget`

**Comment by ctsims** [comment]
@phillipm Can you confirm whether this touches the API stuff at all? I would really like to hold off on refactors of large sets of files until @wpride has that pulled in.

**Comment by phillipm** [comment]
@ctsims I can confirm that there aren't any merge conflicts between the two branches

**Comment by wpride** [comment]
@phillipm good refactor

**Pull request merged**